src/main/reaktivity/protocol.idl

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2016-2019 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+scope protocol
+{
+    option byteorder network;
+
+    scope codec
+    {
+        enum SocksPacketType
+        {
+            CONNECT,
+            BID
+        }
+
+        struct Binary
+        {
+          uint16 length;
+          octets[length] bytes;
+        }
+
+        struct SocksIdentifier
+        {
+          unit8 version;
+          unit16 methods;
+        }
+        struct SocksRequests
+        {
+          unit8 version;
+          unit8 succeeded;
+          unit8 reserved;
+          unit8 addressType;
+          unit32 assignedAddress;
+        }
```

Comment:
The goal is that when we use the flyweights generated from these structures that they layout the bytes in the same way as the SOCKS spec.

Please verify against the spec and I’ll double check later.